### ISSUE TITLE
or_equals and dor_equals do  not add the key to the list of keys

### ISSUE THREAD

**Issue by petdance** [dagolden/Hash-Ordered#8]
Test program to set values and then show the keys & values.

```
#!/usr/bin/perl

use warnings;
use strict;
use 5.010;

use Hash::Ordered;

my $h = Hash::Ordered->new;

$h->set( 'foo', 'is_foo' );
_dump( $h, 'Added first key' );

$h->dor_equals( 'bar', 'is_bar' );
_dump( $h, 'dor_equalsed "bar"' );

$h->set( 'bat', 'is_bat' );
_dump( $h, 'Added third key, via set' );

exit 0;


sub _dump {
    my $h = shift;
    my $msg = shift;

    say $msg;
    say 'Keys:';
    say '    ', join( ', ', $h->keys );
    say 'Values:';
    say '    ', join( ', ', $h->values );
    say '';

    return;
}
```

```
$ perl foo.pl
Added first key
Keys:
    foo
Values:
    is_foo

dor_equalsed "bar"
Keys:
    foo
Values:
    is_foo

Added third key, via set
Keys:
    foo, bat
Values:
    is_foo, is_bat
```

I would expect that after calling `dor_equals` that `keys` would give `foo,bar`.  Looking at the internals of the code, my guess is that it's not doing the 

    CORE::push @{ $self->[_KEYS] }, "$key"; # stringify key

that is in the `set` function.  Also looks like `or_equals` has the same problem.

**Comment by petdance** [comment]
@xdg Any chance we could get an updated release of Hash::Ordered?  We're about to release the software where I discovered the problem, and it would be nice if we could remove the workaround in my code.

**Comment by xdg** [comment]
Shipped 0.13-TRIAL.  Will ship stable after CPAN testers beats on it for a few days.  Thanks for reminding me.
